### PR TITLE
refactor: used a common http client for all the requests

### LINKED ISSUE #1197: Number of goroutines were increasing gradually
# Problem

With every iteration in infinite voting for loop, we see a gradual increase in number of goroutines for few epochs and the count does go down after certain epochs but goes on gradual increase which can cause over usage if CPU if the node is running for a long time.

Check whether every goroutine created should be ensured that its getting closed after its execution.

Check for go routines in a for loop are handled correctly.

Check whether every channel is closed using a defer keyword.

# Causes 
1. Currently node creates separate http client instance for every API and makes request using `client.Do()` which creates a new goroutine on every call 

- Using a common http client for all the requests will help here as the new request can use the existing connections from the connection pool and there will be no need create a new connection everytime in a separate go routine.

2.  `keystore.Accounts()` opens a new goroutine everytime whenever we try to fetch keystore files present in keystore directory. - https://github.com/razor-network/oracle-node/issues/1195

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

There was a gradual increase in number of go routines for every http request that node was making, this was due to new HTTP client being used on every `httpClient.Do()` call. Rather there should be a common HTTP client which handles creating and making requests across all hosts, this will enable the reusing of existing connections from the connection pool.
   
- Used a common http client to avoid opening a new connection on every `client.Do()` call for every API.
- Removed global variable `HttpTimeout` and directly picked it up from config.

Fixes #1197 ,  https://linear.app/interstellar-research/issue/RAZ-771

# How Has This Been Tested?

Ran a node on local and staging environment with 5 min epoch time.

### CODE REVIEW

File: core/constants.go [hunk 1]
```diff
@@ -81,3 +82,8 @@ var HexArrayExtractIndexRegex = `^hexArray\[(\d+)\]$`
 var BatchSize = 1000
 var NumRoutines = 10
 var MaxIterations = 10000000
+
+// Following are the constants used in custom http.Transport configuration for the common HTTP client that we use for all the requests
+
+var HTTPClientMaxIdleConns = 15
```
Comment (ashish10677) [inline core/constants.go:88]: Use const instead of var for these

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: core/constants.go (hunk 1 above)
Comment (Yashk767) [reply core/constants.go:88]: okay I will make this change.
While all the other constants defined in `constants.go` are also defined with `var` keyword. I think we can update to `const` at all these occurrences but in separate PR, not in this. Is this okay @ashish10677 ?

Comment (ashish10677) [reply core/constants.go]: For others, you can open a new PR. For these new variables you've added, change it in this PR itself.

Comment (Yashk767) [reply core/constants.go]: Done in https://github.com/razor-network/oracle-node/pull/1201/commits/897d867f12ee4075cb34e9934e42afacd7518d9c

**Review by ashish10677 — APPROVED** [review]

**Review by rajkharvar — APPROVED** [review]

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**